### PR TITLE
Add jest and basic tests #1

### LINKED ISSUE #1: Unit tests
Add unit tests with Jest. This should be done once the Store/Module API is in a stable state.

- [x] Add jest package and config
- [x] Add tests directory and package.json script
- [x] Add tests covering basic usage

### PR DESCRIPTION
**Pull request by sampullman**
Close #1 

@abemscac I added you as a reviewer, but it's not urgent. Take a look when you get the chance! Once this is merged I'll add the Github CI checks, unless you want to give it a try.

### CODE REVIEW

File: test/store.spec.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+const getDefaultState = (): TestState => ({
+  id: 0,
+  name: '',
+})
+
+const makeTestModule = () =>
+  new Module<TestState, TestStateGetters, TestStateMutations>({
+    state: getDefaultState,
+    getters: {
+      isDefault: (state) => state.id !== 0 || state.name !== '',
```
Comment (abemscac) [inline test/store.spec.ts:30]: If I understand the meaning of `isDefault` correctly, the returned value of this function will actually be `notDefault`, right?

**Comment by abemscac** [comment]
Hi Sam, my VSCode says that `lib/index.ts is not listed within the file list of project tsconfig.spec.json` must be added in the `include` of `tsconfig.spec.json`.
image

I have to include "lib/*.ts" in the `include` array of `tsconfig.spec.json` to make this error gone.
Did you see the same error as I do?

**Comment by abemscac** [comment]
And yes, of course! It would be nice if I can have a chance to try out this Github CI checks.
You could just leave trivial works like this to me, so that you can focus on tasks with higher priority.
Besides, I can learn something new at the same time, which is great.

File: test/store.spec.ts (hunk 1 above)
Comment (sampullman) [reply test/store.spec.ts:30]: Yes, good catch!

**Comment by sampullman** [comment]
> I have to include "lib/*.ts" in the include array of tsconfig.spec.json to make this error gone.

Yes, that was an oversight by me. Sometimes I forget about tsconfig rules, in this case it's because the inherited `include` from `tsconfig.json` is overridden by `tsconfig.spec.json`, but I thought it was merged.

I don't really mind doing things like this every so often, it doesn't take too long. For CI I will leave a comment in the issue with some reference to checks from another project.

**Comment by abemscac** [comment]
Thank you Sam! It works now.
I found there's some problem with the white spacing of `.prettierrc`. I've fixed it in the new commit.

**Comment by abemscac** [comment]
Oh no!
I accidentally clicked merge, too bad 😬.
Sorry about that.

**Comment by sampullman** [comment]
That's ok, no worries

**Pull request merged**